stl/inc/variant

```diff
@@ -1601,6 +1626,51 @@ template <class _Ty>
 using _As_variant = // Deduce variant specialization from a derived type
     decltype(_As_variant_(_STD declval<_Ty>()));
 
+#if _HAS_CXX20
+template <size_t _Size, class _ListOfIndexVectors, class _Ret, class _Callable, class... _Variants>
+constexpr _Ret _Visit_impl(_Callable&& _Obj, _Variants&&... _Args) {
+    constexpr int _Strategy =
+        _Size == 1 ? 0 : _Size <= 4 ? 1 : _Size <= 16 ? 2 : _Size <= 64 ? 3 : _Size <= 256 ? 4 : -1;
+    return _Visit_strategy<_Strategy>::template _Visit<_Ret, _ListOfIndexVectors>(
+        _Variant_visit_index1(0, static_cast<_As_variant<_Variants>&>(_Args)...), static_cast<_Callable&&>(_Obj),
+        static_cast<_As_variant<_Variants>&&>(_Args)...);
+}
+
+template <class _Callable, class... _Variants, class = void_t<_As_variant<_Variants>...>>
+constexpr _Variant_visit_result_t<_Callable, _As_variant<_Variants>...> visit(_Callable&& _Obj, _Variants&&... _Args) {
+    // Invoke _Obj with the contained values of _Args...
+    constexpr auto _Size = _Variant_total_states<_Remove_cvref_t<_As_variant<_Variants>>...>;

```

Comment:
In `_HAS_CXX20` code, you can use C++20 `remove_cvref_t` instead of our `_Ugly` helper `_Remove_cvref_t` (needed for C++17 and earlier). Occurs below. (But perhaps @CaseyCarter would like the C++20 code to remain as similar as possible to the C++17 code.)

Reply:
I think we can use this implementation of `visit` in both C++17 and C++20 mode (see my comment on the `visit` overload outside the `_HAS_CXX20` block), in which case these should certainly stay `_Remove_cvref_t`s. I think I'd prefer to keep the two overloads consistent where possible so the true differences stand out, so I'd leave the other overload alone as well, but I could easily be convinced otherwise if there were a reason to prefer `remove_cvref_t` to `_Remove_cvref_t` other than simply "the former is the standard's name".

Reply:
I also agree that if we use the vanilla `std::visit()` in both C++17 and C++20 then we should use `_Remove_cvref_t` in `std::visit<R>()` to keep the consistency between not only the two `visit` variations, but also among the file.